C++/Navio/Common/Ublox.cpp

```diff
@@ -346,6 +347,55 @@ int Ublox::testConnection()
     return 0;
 }
 
+int Ublox::configureSolutionRate(std::uint16_t meas_rate,
+                           std::uint16_t nav_rate,
+                           std::uint16_t timeref)
+{
+    CfgNavRate msg;
+    msg.measure_rate = meas_rate;
+    msg.nav_rate     = nav_rate;
+    msg.timeref      = timeref;
+
+    _sendMessage(CLASS_CFG, MSG_CFG_RATE, &msg, sizeof(CfgNavRate));
+}
+
+int Ublox::_sendMessage(std::uint8_t msg_class, std::uint8_t msg_id, void *msg, std::uint16_t size)
+{
+    unsigned char buffer[buffer_length];
+
+    UbxHeader header;
+    header.preamble1 = PREAMBLE1;
+    header.preamble2 = PREAMBLE2;
+    header.msg_class = msg_class;
+    header.msg_id    = msg_id;
+    header.length    = size;
+
+    int offset = _spliceMemory(buffer, &header, sizeof(UbxHeader));
+    offset = _spliceMemory(buffer, msg, size, offset);
+
+    auto checksum = _calculateCheckSum(buffer, offset);
+    offset = _spliceMemory(buffer, &checksum, sizeof(CheckSum), offset);
+
+    return SPIdev::transfer(spi_device_name.c_str(), buffer, nullptr, offset, 200000);
+}
+
+int Ublox::_spliceMemory(unsigned char *dest, const void *src, std::size_t size, int dest_offset)
+{
+    std::memmove(dest + dest_offset, src, size);
+    return dest_offset + size;
+}
+
+Ublox::CheckSum Ublox::_calculateCheckSum(unsigned char message[], std::size_t size) {
```

Comment:
`unsigned char *message` is more conventional

Reply:
Fixed